mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,7 +114,7 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val bucketizer = new Bucketizer(uid).setSplits(splits)
+    val bucketizer = new Bucketizer(uid).setSplits(splits.distinct.sorted)
```

Comment:
I'm more prefer to leave warning log for this condition. Since if users set ```numBuckets = 10```, and then get buckets with less than the ```numBuckets```, it will confuse users. What about your thoughts? @srowen